### PR TITLE
Nc news 09

### PR DESCRIPTION
**Pull request by crissu91**

### CODE REVIEW

File: src/components/AllTopics.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    useEffect(()=>{
+        getAllTopics()
+        .then((data)=>{
+            setTopics(data)
+            setIsLoading(false)
+        })
+        .catch((err) =>{
+            setApiError(err)
+        })
+    },[])
```
Comment (NRMSMITH) [inline src/components/AllTopics.jsx:20]: What is this useEffect dependent on?

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Have a think about what your useEffect is dependent on :)

**Review by mezzDavies — CHANGES_REQUESTED** [review]
Currently when the user clicks on a topic the url changes to this - `/articles?topic=coding` ideally we want it to be `/topics/coding `- this makes much more sense to the user as we are moving forwards from the `topics` page.  Put another way the endpoint we use to access the API doesn't have to be the url that is shown on the front end. Please implement this. 🙂

Having spoken to Niamh I think this is what she was getting at with her comment earlier - you do not need topics in the dependency array 👍

**Comment by NRMSMITH** [comment]
This looks brill Cristina! 🚀 Apologies about my misleading comment earlier! 😓 That's my bad!! 

Merge away 😄

**Pull request merged**